src/transformers/models/graphormer/collating_graphormer.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) Microsoft Corporation and HuggingFace
+# Licensed under the MIT License.
+
+from typing import Any, Dict, List, Mapping
+
+import numpy as np
+import torch
+
+import pyximport
```

Comment:
Not 100% sure this is part of the standard library. If this is an extra dep, then this import needs to be guarded inside an `if is_xxx_available()`.

Reply:
pyximport comes from Cython > is it OK to have as a dependency? 

I has initially ported the code to Python but it was so slooow (several orders of magnitude slower)

Reply:
It is okay to have a soft dependency yes. You just need to add an `is_cython_available` function in the import utils, and protect the import here to be under a test `if is_cython_available()`. Then in the init of your data collator, the first line should be `requires_backend(["cython"])` so an error is thrown if it's not installed (if it's possible to have not packaged with python).